### DISCUSSION TITLE
Connect wallet in magento 2

### DISCUSSION THREAD

**Discussion by mohitvhits** [WalletConnect/walletconnect-monorepo#4186] [Developer Support]
hello dear,

we will try to connect a web3.0 wallet connection in Magento 2 technology. but we can't connect the wallet connection. get the problem is those js packages require lots of all other js from node_modules. so please suggest me and help. i am waiting for your reply.

thanks regard

**Comment by RohitGohelVH** [comment]
Yes I have also tried to connect the wallet using the javascript solution given by this link 

https://docs.walletconnect.com/web3modal/javascript/about

but we can not get any solution because they need too many javascript dependencies and all javascript file dependencies have come from the node_module we can not add into the Magento so please help us to do this thing into the magento 2 payment method.

**Comment by vinaywebcom** [comment]
One way is to use headless magento with react frontend.

**Reply by RohitGohelVH** [reply]
yes, this is the way but we are trying to connect to the community edition how can we achieve this, most magento sites are working into the community edition or enterprise edition so this solution works only for headless magento not for all.
